consensus/pbft/core.go

```diff
@@ -34,14 +34,14 @@ func (c pbftCore) currentPrimary() uint {
 	return c.view % c.n
 }
 
-// TODO (pbft): are there different quorums? this one is for the prepare messages.
 func (c pbftCore) prepareQuorum() uint {
-	// TODO (pbft): Check if we should use 2f here or, as in fabric - (n+f+2)/2?
+	// TODO (pbft): Not sure about this quorum - is it 2f total or 2f prepares from OTHER replicas?
```

Comment:
2f+1 total consensus, BUT we need 3f+1 formation. i.e. 4 nodes in formation to be tolerant of 2f+1

Reply:
I was thinking more of this one specific condition. The PBFT paper says:

```
We define the predicate prepared(m,v,n,i) to be true if and only if replica i has inserted in its log: the request m,
a pre-prepare for m in view v with sequence number n, and 2f prepares from different backups that match
the pre-prepare. The replicas verify whether the prepares match the pre-prepare by checking that they have the
same view, sequence number, and digest.
```

So while e.g. the quorum for `commit` condition, or `view-change` is `2f+1`, I actually believe for `prepare` condition it's `2f`.
The reason being - the primary that pre-prepared it also counts as effectively voting for it, so with 2f replicas agreeing, that amounts to `2f+1`.

That's my understanding at least. So for other conditions the quorum we expect is 2f+1, but for the `prepare` condition specifically it's `2f`.